src/engine.rs

```diff
@@ -348,13 +361,19 @@ impl Engine {
 
     /// Cleans the collector and store owned by the engine.
     pub fn cleanup(&self) -> Result<(), Failed> {
-        self.store.cleanup(self.collector.cleanup())
+        self.store.cleanup(
+            self.collector.as_ref().map(|collector| {
```

Comment:
Any reason this can't be written as `as_ref().map(Collector::cleanup)`, to be shorter but moreover consistent with line 343?

Reply:
Apparently, no reason. I am surprised Clippy didn’t complain.